harvest-legacy/src/main/java/gov/nasa/pds/harvest/search/util/XMLExtractor.java

```diff
@@ -286,4 +289,81 @@ public List<String> getAttributeValuesFromItem(String expression, Object item)
         }
         return vals;
     }
+
+    /**
+     * Gets the values of the given expression as JSON strings.
+     * Each matching node is converted to a JSON object and returned as a string.
+     *
+     * @param expression An XPath expression.
+     *
+     * @return A list of JSON strings, one for each matching node.
+     *
+     * @throws XPathExpressionException If the given expression was malformed.
+     */
+    public List<String> getValuesAsJsonFromDoc(String expression)
+    throws XPathExpressionException {
+        return getValuesAsJsonFromItem(expression, xml);
+    }
+
+    /**
+     * Gets the values of the given expression as JSON strings.
+     * Each matching node is converted to a JSON object and returned as a string.
+     *
+     * @param expression An XPath expression.
+     * @param item The starting point from which to evaluate the XPath expression.
+     *
+     * @return A list of JSON strings, one for each matching node.
+     *
+     * @throws XPathExpressionException If the given expression was malformed.
+     */
+    public List<String> getValuesAsJsonFromItem(String expression, Object item)
+    throws XPathExpressionException {
+        List<String> jsonStrings = new ArrayList<String>();
+        List<TinyElementImpl> nList = (List<TinyElementImpl>) xpath.evaluate(
+            expression, item, XPathConstants.NODESET);
+
+        if (nList != null) {
+            for (int i = 0, sz = nList.size(); i < sz; i++) {
+                TinyElementImpl node = nList.get(i);
+                try {
+                    // Convert the node to an XML string
+                    String xmlString = nodeToString(node);
+
+                    // Convert XML to JSON using org.json library
+                    JSONObject jsonObject = XML.toJSONObject(xmlString);
+
+                    // Add the JSON string to the result list
+                    jsonStrings.add(jsonObject.toString());
+                } catch (Exception e) {

```

Comment:
Better to catch and rethrow `RuntimeException` and then handle `Exception` thanks to the broken Java exception hierarchy; [see this article by yours truly](https://github.com/NASA-PDS/nasa-pds.github.io/wiki/Java-Exceptions-Gotcha) 😉